client/components/layout/AsideMenu/AsideMenu.tsx

```diff
@@ -34,6 +34,17 @@ function AsideMenu({
     } else {
       document.body.style.overflow = 'initial';
     }
+
+    const handleEsc = (event) => {
```

Comment:
Let's extract this into a separate hook, so we can easily re-use this code among components.

Have a look at `useOutsideClick` for inspiration.